### ISSUE TITLE
Bug or feature? Mapping id the same within a connected set?

### ISSUE THREAD

**Issue by kerimoyle** [cellml/libcellml#629]
Is this a bug, or is it expected?  Two individual mappings in one component connect to the same variable in another, and they then share the same mapping id?
v1 -> v2a  } These two are the same?
v1 -> v2b  }
v1 -> v3 ... but this one is different
Any thoughts @hsorby @agarny @nickerso ?

```
TEST(Connection, mappingId)
{
    const std::string in = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
                           "<model xmlns=\"http://www.cellml.org/cellml/2.0#\"  name=\"everything\" >\n"
                           "  <component name=\"c1\" >\n"
                           "    <variable name=\"v1\" units=\"units2\" interface=\"private\"/>\n"
                           "  </component>\n"
                           "  <component name=\"c2\">\n"
                           "    <variable name=\"v2a\" units=\"units2\" interface=\"public\"/>\n"
                           "    <variable name=\"v2b\" units=\"units2\" interface=\"public\"/>\n"
                           "  </component>\n"
                           "  <component name=\"c3\">\n"
                           "    <variable name=\"v3\" units=\"units2\" interface=\"public\"/>\n"
                           "  </component>\n"
                           "  <connection component_1=\"c1\" component_2=\"c2\">\n"
                           "    <map_variables variable_1=\"v1\" variable_2=\"v2a\" id=\"id12a\"/>\n"
                           "    <map_variables variable_1=\"v1\" variable_2=\"v2b\" id=\"id12b\"/>\n"
                           "  </connection>\n"
                           "  <connection component_1=\"c1\" component_2=\"c3\">\n"
                           "    <map_variables variable_1=\"v1\" variable_2=\"v3\" id=\"id13\"/>\n"
                           "  </connection>\n"

                           "</model>\n";
    auto parser = libcellml::Parser::create();
    auto model = parser->parseModel(in);
    EXPECT_EQ("id12a",
              libcellml::Variable::equivalenceMappingId(
                  model->component("c1")->variable("v1"),
                  model->component("c2")->variable("v2a")));  // <<< Fails because id is "id12b"
    EXPECT_EQ("id12b",
              libcellml::Variable::equivalenceMappingId(
                  model->component("c1")->variable("v1"),
                  model->component("c2")->variable("v2b"))); 
    EXPECT_EQ("id13",
              libcellml::Variable::equivalenceMappingId(
                  model->component("c1")->variable("v1"),
                  model->component("c3")->variable("v3")));
}
```

**Comment by kerimoyle** [comment]
(and on another note, there are no tests which use `Variable::equivalenceMappingId` directly ... is that not a problem? I would have expected some kind of coverage flag if API functions weren't called in tests?)

**Comment by agarny** [comment]
Your `v2a` and `v2b` variables are distinct, so although they are both equivalent to `v1`, I would expect the id to be different. IOW, it's a bug for me.

**Comment by kerimoyle** [comment]
Even though they're both in the same component?  That's allowed, right? The bug doesn't appear on the connection to component3.

**Comment by agarny** [comment]
It's true that to define `v2a` and `v2b` in the same component and then to have both of them to be equivalent to `v1` is a bit weird, but the CellML code is clearly valid. So, when requesting the id for the equivalence mapping between `v1` and `v2a`, I would expect to get `id12a`.

IOW, you have only yourself to blame for coming up with some twisted cases. 🤪

**Comment by kerimoyle** [comment]
Haha! Twisted is my speciality ... though in fairness, this did come from modifying an existing test ... we just haven't tested `equivalenceMappingId` before :)

**Comment by kerimoyle** [comment]
For the record, I will fix this in PR #630.

**Comment by kerimoyle** [comment]
The bug is in the parser - it doesn't reset the id of mappings when reading more than one in the same connection block.  Will push the fix soon.
